### DISCUSSION TITLE
Single sign on for web testing on iOS platform

### DISCUSSION THREAD

**Discussion by luprochazka-cen63872** [appium/appium#19653] [Q&A]
Hi, I use RobotFramework with Browser library (Playwright) for desktop apps. I am able to add credentials as a argument when I open new context of browser for SSO. With this concept I can easily switch between many users and test various behavior. 

`New Context  viewport={'width': 1920, 'height': 960}  httpCredentials={'username': '$USERNAME', 'password': '$PASSWORD'`

Is there anyway how can I set credentials for a Appium web tests?

I start a test like this:

`Open Application  ${APPIUM_SERVER_URL}  platformName=${PLATFORMA}  deviceName=${NAZEV_ZARIZENI}  udid=${UDID} browserName=safari  automationName=XCUITest`

Thank you.

**Comment by KazuCocoa** [comment]
No, it doesn't have such a feature.

It looks like the command depends on `Emulation.setAuthCredentials`. It exists only Playwright customized WebKit ( https://github.com/microsoft/playwright/blob/268942037a054f7259ecee178645c9067a8ee2e1/browser_patches/webkit/patches/bootstrap.diff ) only, it is not in the vanilla WebKit (usually used for Safari), thus Appium cannot support it with few works at least. https://github.com/appium/appium-remote-debugger/blob/master/lib/protocol/index.js
